jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -97,19 +94,19 @@ class QueryEditorDialog(
         queryStartEndDate = (if (funDetails.absoluteTimeSelected) {
             getAbsoluteTime(funDetails.startDateAbsolute, funDetails.endDateAbsolute)
         } else {
-            getRelativeTime(funDetails.relativeTimeUnit, funDetails.relativeTimeNumber.toLong())
+            getRelativeTime(relativeTimeUnit[funDetails.relativeTimeUnit], funDetails.relativeTimeNumber.toLong())
         })
-        val queryStartDate = queryStartEndDate.startDate.toEpochMilli() / 1000
-        val queryEndDate = queryStartEndDate.endDate.toEpochMilli() / 1000
-        val query = if (funDetails.queryingLogsSelected) {
+        val queryStartDate = queryStartEndDate.startDate
+        val queryEndDate = queryStartEndDate.endDate
+        val query = if (funDetails.enterQuery) {
             funDetails.query } else {
             getFilterQuery(funDetails.searchTerm)
         }
         close(OK_EXIT_CODE)
-        queryingLogGroupApiCall.executeStartQuery(queryEndDate, funDetails.logGroupName, query, queryStartDate)
+        queryingLogGroupApiCall.executeStartQuery(queryStartDate, queryEndDate, funDetails.logGroupName, query)
     }
     private fun getFunctionDetails(): QueryDetails = QueryDetails(
-        logGroupName = lGroupName,
+        logGroupName = logGroupNames,
```

Comment:
Is this the correct place to put this method? I feel like it makes more sense to be in `QueryEditor.kt`. What are we doing for our other views?

Reply:
Later, when the table is implemented and multiple log groups can be selected, the array will be created in QueryEditorDialog so the logGroupNames will be updated here. I referred to the EditFunctionDialog class where a similar function is implemented for this view.